packages/react-html/src/tests/serializer.test.tsx

```diff
@@ -23,4 +24,36 @@ describe('useSerialized', () => {
       `<script type="text/json" data-serialized-id="foo">"foo_value"</script>`,
     );
   });
+
+  it('gets the latest serialization value on re-render', () => {
+    const key = 'serialized-key';
+    const defaultValue = 'default-value';
+
+    function MockComponent() {
+      const [value] = useSerialized<string>(key);
+      return <div>{value || defaultValue}</div>;
+    }
+
+    const manager = new HtmlManager();
+
+    const mount = createMount({
+      render: (element) => (
+        <HtmlContext.Provider value={manager}>{element}</HtmlContext.Provider>
+      ),
+    });
+
+    const wrapper = mount(<MockComponent />);
+
+    expect(manager.getSerialization(key)).toBeUndefined();
+    expect(wrapper.text()).toBe(defaultValue);
+
+    const newValue = 'new-value';
+    manager.setSerialization(key, newValue);
+    // forces re-render
+    wrapper.setProps({bar: true});
```

Comment:
Rather than change a prop to trigger re-render as a side effect, we should aim to use `act` here to show that setting the serialization will have the desired effect:

```suggestion
    wrapper.act(() => {
      manager.setSerialization(key, newValue);
    });
```

However I suspect that won't work, because I don't think we've configured the `manager` as as a mutable store of value to indicate to React that a new render is necessary.

What this means, is that we'll get some tearing where the serialization value can be updated, but it doesn't cause a render. Then when a render happens for other reasons, it will use the new updated value. Is that by intent?

Reply:
You're right that it won't work in an `act` since it doesn't trigger anything by itself. 

Serialized values never change during the lifetime of the app, it's mostly useful in a testing environment.

By default, it should have worked, but the `useMemo` had the unintended effect of hiding any change to the serialization map in the manager. So any code using the manager _was_ getting the latest values, only React components using the `useSerialized` hook were not getting the latest values.